src/interface/lazy.jl

```diff
@@ -7,12 +7,13 @@ const AbstractArrayOrBroadcasted = Union{AbstractArray,Broadcasted}
 mutable struct LazyTensor{T, N}
     data
     extrude::NTuple{N, Bool}
+    shape::NTuple{N, Int}
```

Comment:
For better or worse, Finch array dimensions may use different types independently. So you could have a matrix with row dimension `UInt8` and column dimension `BigInt`. The way we handle this is to parameterize the type of LazyTensor based on the type of the dimensions: 
```
mutable struct LazyTensor{T, N, Shape <: Tuple}
    data
    extrude::NTuple{N, Bool}
    shape::Shape
    fill_value::T
end
```
Note that the dimension tuple is not constrained to be length `N` with the type annotations. That's okay, we can check that it's correct when we construct the LazyTensor.